include/oneapi/dpl/internal/random_impl/philox_engine.h

```diff
@@ -0,0 +1,544 @@
+// -*- C++ -*-
+//===-- philox-engine.h ------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Public header file provides implementation for Philox Engine
+
+#ifndef _ONEDPL_PHILOX_ENGINE_H
+#define _ONEDPL_PHILOX_ENGINE_H
+
+#include <cstdint>
+#include <utility>
+#include <type_traits>
+#include <limits>
+#include <array>
+#include <iostream>
+#include <algorithm>
+
+#include "random_common.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace experimental
+{
+
+template <typename _UIntType, std::size_t _w, std::size_t _n, std::size_t _r,
+          internal::element_type_t<_UIntType>... _consts>
+class philox_engine;
+
+template <typename __CharT, typename __Traits, typename __UIntType, std::size_t __w, std::size_t __n, std::size_t __r,
+          internal::element_type_t<__UIntType>... __consts>
+std::basic_ostream<__CharT, __Traits>&
+operator<<(std::basic_ostream<__CharT, __Traits>&, const philox_engine<__UIntType, __w, __n, __r, __consts...>&);
+
+template <typename __UIntType, std::size_t __w, std::size_t __n, std::size_t __r, internal::element_type_t<__UIntType>... __consts>
+const sycl::stream&
+operator<<(const sycl::stream&, const philox_engine<__UIntType, __w, __n, __r, __consts...>&);
+
+template <typename __CharT, typename __Traits, typename __UIntType, std::size_t __w, std::size_t __n, std::size_t __r,
+          internal::element_type_t<__UIntType>... __consts>
+std::basic_istream<__CharT, __Traits>&
+operator>>(std::basic_istream<__CharT, __Traits>&, philox_engine<__UIntType, __w, __n, __r, __consts...>&);
+
+template <typename _UIntType, std::size_t _w, std::size_t _n, std::size_t _r,
+          internal::element_type_t<_UIntType>... _consts>
+class philox_engine
+{
+    /* The size of the consts arrays */
+    static constexpr std::size_t __array_size = _n / 2;
+
+    /* Methods for unpacking variadic of constants into two arrays */
+    template <typename _Array, std::size_t... _Is>
+    static constexpr auto
+    get_even_elm_array(_Array __inp, std::index_sequence<_Is...>)
```

Comment:
minor: I would suggest get_even_elm_array -> get_even_elem_array or get_even_element_array

Reply:
Renamed to `get_even_element_array` and `get_odd_element_array`